components/dropdown/test/dropdown-helper.js

```diff
@@ -0,0 +1,29 @@
+const getEvent = (page, selector, name) => {
+	return page.$eval(selector, (elem, name) => {
+		return new Promise((resolve) => {
+			elem.addEventListener(name, resolve, { once: true });
+		});
+	}, name);
+};
+
+const asyncSetTimeout = (delay) => {
+	return new Promise(resolve => setTimeout(resolve, delay));
+};
+
+module.exports = {
+
+	getOpenEvent(page, selector) {
+		return getEvent(page, selector, 'd2l-dropdown-open');
+	},
+
+	async open(page, selector) {
+		const openEvent = this.getOpenEvent(page, selector);
+		await page.$eval(selector, (dropdown) => {
+			dropdown.toggleOpen();
+		});
+		// 300ms fade-in animation on open
+		await asyncSetTimeout(300);
```

Comment:
Normally I would handle the animation by waiting for the animation:
```
await page.$eval(selector, (dropdown) => {
    return new Promise((resolve) => {
        dropdown.querySelector('d2l-dropdown-content').addEventListener('animationend', () => resolve(), { once: true });
        dropdown.toggleOpen();
    });
});
```
Aside, if this approach is used with `transitionend` events, one must be careful to wait for all the properties to animate... the event will fire once for each property being transitioned, which can obviously have different durations.